### DISCUSSION TITLE
Any plans to develop a stream viewer

### DISCUSSION THREAD

**Discussion by thaoula** [rabbitmq/rabbitmq-server#8730] [Other] [answered]
Hi,

We have just started to use streams. We are using Nodejs via the Rascal library, the lack of a stream protocol driver means we need must use amqp 0.91.

The management console does not allow you to view messages within a stream. So, I was wondering if I was missing something or whether there was another way to view a streams contents.

During the development process it would really be beneficial to be able to view the messages to validate what is occurring.

If not, are there any plans to develop a stream viewer that can iterate over the messages in a stream. 

Regards,
Tarek

**Comment by michaelklishin** [comment]
Management UI does not allow you to browse queues either, the best you can do is a consume-then-requeue loop is modifies the state of the queue (the redelivered message state, and while the loop is in progress, the head of the queue).

But stream consumption is non-destructive, so it should be fairly trivial to develop a CLI tool that iterates over the stream without affecting its state.

RabbitMQ is open source software. If you need a stream viewer, you can build one.

**Comment by thaoula** [comment]
Hi @michaelklishin, 

Really appreciate your prompt response. The exact reason I was asking the question was actually highlighted in your response.

The management tool allows you call get messages even though it is a destructive action but it provides a workaround by allowing you to specify nack which leads to issue you mentioned.

Considering the stream where the action is not destructive and more natural what was the reason get message was not supported for them.

So my real question is will get messages ever support streams? if not, is there any gotchas such as paging etc if using amqp.

Regards,
Tarek

**Reply by michaelklishin** [reply, marked as answer]
Management UI does not have a complete context that stream consumers have, so we will see.

Again, anyone can use their favorite RabbitMQ stream client to develop a small CLI-based stream browser (Web-based would be trickier to get right).

**Comment by thaoula** [comment]
For now, we are going to push the contents into MongoDB to allow developers to view the contents and debug issues.
